### PR TITLE
[MOB-2026] auth failure delegate

### PR DESCRIPTION
**Pull request by roninopf**

### CODE REVIEW

File: swift-sdk/Internal/NetworkHelper.swift [hunk 1]
```diff
@@ -10,10 +10,14 @@ typealias SendRequestValue = [AnyHashable: Any]
 struct SendRequestError: Error {
     let reason: String?
     let data: Data?
+    let responseCode: Int?
```
Comment (tapashmajumder) [inline swift-sdk/Internal/NetworkHelper.swift:13]: It is named `httpStatusCode` in offline-events branch.

**Comment by tapashmajumder** [comment]
We have added the variable `responseCode` but I don't see where we are setting it in SendRequestError. Can you please take a look at offline-events branch?

**Comment by roninopf** [comment]
> We have added the variable `responseCode` but I don't see where we are setting it in SendRequestError. Can you please take a look at offline-events branch?

It was used the same way `httpStatusCode` is; but I'm only using it in the 401. But, I'll reconcile the version on the `offline-events` branch into here instead.

File: swift-sdk/Constants.swift [hunk 2]
```diff
@@ -195,6 +195,10 @@ public enum JsonKey: String, JsonKeyRepresentable {
         static let defaultAction = "defaultAction"
     }
     
+    public enum Response {
+        static let code = "code"
```
Comment (tapashmajumder) [inline swift-sdk/Constants.swift:199]:
```suggestion
        static let iterableCode = "iterableCode"
```

**Review by tapashmajumder — APPROVED** [review]
+1

**Pull request merged**